name: "throw"
type: :function
description: "throw an error with given message"
module: "Exceptions"
returns: [
                :nothing
        ]
line: 45
source: "https://github.com/arturo-lang/arturo/blob/v0.9.83/src/library/Exceptions.nim#L45"
examples: [
                {
                        err: try -> throw "Page not found"
                        err\kind
                        ; => Generic Error
                        err\message
                        ; => Page not found
                        
                        ; or you can alternatively use custom errorKind
                        
                        pageNotFound: to :errorKind "404: Page not Found"
                        
                        err: try -> throw.as: pageNotFound "Seems that the page does not exist"
                        err\kind
                        ; => 404: Page not Found
                        err\message
                        ; => Seems that the page does not exist
                        
                        ; Or even use the :errorKind's label as the message itself
                        
                        err: try -> throw pageNotFound
                        err\kind
                        ; => 404: Page not Found
                        err\message
                        ; => 404: Page not Found
                }
        ]
attributes: #[
        as: #[
                types: [
                                :errorkind
                        ]
                description: "consider the error as one of given subtype"
                ]
        ]
arguments: #[
        message: [
                        :string
                         :errorkind
                ]
        ]
related: [
                "Exceptions/try"
                 "Exceptions/assertionError"
                 "Exceptions/indexError"
                 "Exceptions/arithmeticError"
        ]